cmd/ncproxy/ncproxy.go

```diff
@@ -222,31 +228,35 @@ func (s *grpcService) CreateNetwork(ctx context.Context, req *ncproxygrpc.Create
 		return nil, status.Errorf(codes.FailedPrecondition, "network with name %q already exists", req.Name)
 	}
 
-	// Get the layer ID from the external switch. HNS will create a transparent network for
-	// any external switch that is created not through HNS so this is what we're
-	// searching for here. If the network exists, the vSwitch with this name exists.
-	extSwitch, err := hcn.GetNetworkByName(req.SwitchName)
-	if err != nil {
-		if _, ok := err.(hcn.NetworkNotFoundError); ok {
-			return nil, status.Errorf(codes.NotFound, "no network/switch with name `%s` found", req.SwitchName)
+	policies := []hcn.NetworkPolicy{}
+	if req.SwitchName != "" {
```

Comment:
I added this change to allow us to make other network types than just transparent as was the previous limitation. This is also needed for the tests since we can't make transparent networks without an adapter.

Reply:
That's fine, I didn't like how I made this required to start with.